tests/ha/control_pod/test_control_pod_soft_failures.py

```diff
@@ -0,0 +1,617 @@
+        for pod in self.ctrl_pod_list:
+            LOGGER.info("Step 2.1: Create soft failure on pod %s", pod)
+            resp = self.node_master_list[0].kill_process_in_container(
+                pod_name=pod, container_name=self.container_name, pid=self.pid, safe_kill=True)
+            LOGGER.info("Response: %s", resp)
+            LOGGER.info("Step 2.1: Created soft failure")
+            LOGGER.info("Step 2.2: Create an IAM user")
+            user = self.ha_obj.create_iam_user_with_header(num_users=1, header=self.header)
+            LOGGER.debug("Created user: %s", user)
+            if pod == self.ctrl_pod_list[-1]:
+                assert_utils.assert_equal(user, None, "Expected failure in IAM user creation")
+                LOGGER.info("Step 2.2: Failed IAM user creation as expected")
+            else:
+                assert_utils.assert_not_equal(user, None, "Expected successful IAM user creation")
+                self.s3_clean.update(user)
+                self.uids.append(user[list(user.keys())[0]]["user_name"])
+                LOGGER.info("Step 2.2: Successfully created IAM user")
+            d_user = del_users.pop(list(del_users.keys())[0])
+            LOGGER.info("Step 2.3: Delete iam user %s", d_user["user_name"])
+            d_user_dict = {d_user["user_name"]: d_user}
+            resp = self.ha_obj.delete_iam_user_with_header(user=d_user_dict, header=self.header)
+            if pod == self.ctrl_pod_list[-1]:
+                assert_utils.assert_false(resp[0], "Expected failure in IAM user deletion")
+                LOGGER.info("Step 2.3: Failed IAM user deletion as expected")
+                self.uids.append(d_user["user_name"])
+                self.s3_clean.update(d_user_dict)
+            else:
+                assert_utils.assert_true(resp[0], "Expected successful IAM user deletion")
+                LOGGER.info("Step 2.3: Successfully deleted IAM user")
+
+            LOGGER.info("Step 2.4: Verify if %s IAM users are persistent across control pod "
+                        "soft failures", self.uids)
+            for user in self.uids:
+                resp = self.rest_iam_user.get_iam_user(user)
+                assert_utils.assert_equal(int(resp.status_code), HTTPStatus.OK.value,
+                                          f"Couldn't find user {user} after control pod "
+                                          "failover")
+                LOGGER.info("Step 2.4: User %s is persistent: %s", user, resp)
+        LOGGER.info("Step 2: Created soft failures and performed IAM user creation")
+
+        LOGGER.info("Step 3: Checking cluster status")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        assert_utils.assert_false(resp[0], resp[1])
+        LOGGER.info("Step 3: Cluster status should be bad as control pods are in failed state")
+
+        LOGGER.info("Step 4: Perform WRITE-READ-Verify on user %s", self.uids[0])
```

Comment:
so even when cluster not in good state we are expecting IOs to run.
Jsut imagining a case where bcoz of some reason some rgw or ioservice went down, so cluster will be offline as expected but then won't this fail as well? So don't we rather want to use poll_cluster to check cluster status after step 2 and once online then only try for IOs?

Reply:
No, added IOs here because even if the cluster is in failed state due to csm-agent, it should not affect IOs. So if IOs are failing then thats an issue.